### ISSUE TITLE
Mypy doesn't catch positional argument error if function has a decorator

### ISSUE THREAD

**Issue by ajhynes7** [python/mypy#14417]
**Bug Report**

Mypy doesn't raise an error when a function call is missing a positional argument, if that function has a decorator on it.

**To Reproduce**

```python
from typing import Callable


def print_hey(func: Callable):

    def wrapper(*args, **kwargs):
        print("hey")
        return func(*args, **kwargs)

    return wrapper


@print_hey
def is_even(x: int):

    return x % 2 == 0


def test_is_even() -> None:

    is_even()
```

https://mypy-play.net/?mypy=latest&python=3.11&gist=4bcd971cb56fa25c7192243df202353f

**Expected Behavior**

I'd expect an mypy error to be raised here, because calling the function without the positional argument results in a runtime exception.

`pyright` raises the following error on the provided example:

```
 error: Argument missing for parameter "x" 
```

**Actual Behavior**

No errors are found.

**Your Environment**

- Mypy version used: 0.991
- Mypy command-line flags: None
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: Python 3.11.0

**Comment by JelleZijlstra** [comment]
Your decorator is untyped, so the type of `is_even` becomes Any. See https://mypy.readthedocs.io/en/stable/generics.html#declaring-decorators.

**Comment by ajhynes7** [comment]
I see, thanks for linking that relevant documentation.
